From mingo@elte.hu  Tue Mar  4 12:49:48 2008
From: Ingo Molnar <mingo@elte.hu>
Date: Tue, 4 Mar 2008 14:54:47 +0100
Subject: sysfs: CONFIG_SYSFS_DEPRECATED fix
To: Greg KH <gregkh@suse.de>
Cc: "Zhang, Rui" <rui.zhang@intel.com>, kay.sievers@vrfy.org, Andrew Morton <akpm@linux-foundation.org>, "Rafael J. Wysocki" <rjw@sisk.pl>, Linus Torvalds <torvalds@linux-foundation.org>, Alexey Dobriyan <adobriyan@sw.ru>
Message-ID: <20080304135447.GD32383@elte.hu>
Content-Disposition: inline


From: Ingo Molnar <mingo@elte.hu>

CONFIG_SYSFS_DEPRECATED=y changed its meaning recently and causes
regressions in working setups that had SYSFS_DEPRECATED disabled.

so rename it to SYSFS_DEPRECATED_V2 so that testers pick up the new
default via 'make oldconfig', even if their old .config's disabled
CONFIG_SYSFS_DEPRECATED ...

Signed-off-by: Ingo Molnar <mingo@elte.hu>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 init/Kconfig |    4 ++++
 1 file changed, 4 insertions(+)

--- a/init/Kconfig
+++ b/init/Kconfig
@@ -367,9 +367,13 @@ config RESOURCE_COUNTERS
 	depends on CGROUPS
 
 config SYSFS_DEPRECATED
+	bool
+
+config SYSFS_DEPRECATED_V2
 	bool "Create deprecated sysfs files"
 	depends on SYSFS
 	default y
+	select SYSFS_DEPRECATED
 	help
 	  This option creates deprecated symlinks such as the
 	  "device"-link, the <subsystem>:<name>-link, and the
